File: FamlyAndroidEdition/app/src/main/java/com/mack/john/famly_androidedition/objects/post/Post.java
package com.mack.john.famly_androidedition.objects.post;

import android.graphics.Bitmap;

import com.mack.john.famly_androidedition.objects.account.Account;
import com.mack.john.famly_androidedition.objects.account.profile.Profile;
import com.mack.john.famly_androidedition.utils.AccountUtils;

import java.io.Serializable;
import java.util.Date;

public class Post implements Serializable, Comparable<Post> {



    // Class properties
    String postMessage;
    Date timeStamp;
    String posterId;
    String posterName;
    boolean hasImage;



    // Constructor
    public Post(String postMessage, Date timeStamp, String posterId, String posterName, boolean hasImage) {
        this.postMessage = postMessage;
        this.timeStamp = timeStamp;
        this.posterId = posterId;
        this.posterName = posterName;
        this.hasImage = hasImage;
    }



    // System generated methods
    @Override
    public int compareTo(Post post) {
        return getTimeStamp().compareTo(post.getTimeStamp());
    }




    // Getters / Setters
    public String getPostMessage() {
        return postMessage;
    }

    public void setPostMessage(String postMessage) {
        this.postMessage = postMessage;
    }

    public Date getTimeStamp() {
        return timeStamp;
    }

    public void setTimeStamp(Date timeStamp) {
        this.timeStamp = timeStamp;
    }

    public String getPosterId() {
        return posterId;
    }

    public void setPosterId(String posterId) {
        this.posterId = posterId;
    }

    public String getPosterName() {
        return posterName;
    }

    public void setPosterName(String posterName) {
        this.posterName = posterName;
    }

    public boolean getHasImage() {
        return hasImage;
    }

    public void setHasImage(boolean hasImage) {
        this.hasImage = hasImage;
    }




    // Custom methods
    public String getPostId() {
        return posterId + timeStamp.getTime();
    }
}

File: Fam.ly - iOS Edition/Fam.ly - iOS Edition/Objects/Post/Post.swift
//
//  File.swift
//  Fam.ly - iOS Edition
//
//  Created by Johnny Mack on 6/3/19.
//  Copyright © 2019 John Mack. All rights reserved.
//

import Foundation

public class Post: Codable {
    
    
    
    // Class properties
    var postMessage: String
    var timeStamp: Date
    var posterId: String
    var posterName: String
    var hasImage = false
    
    
    
    // Constructor
    init(PostMessage postMessage: String, TimeStamp timeStamp: Date, PosterId posterId: String, PosterName posterName: String, HasImage hasImage: Bool) {
        self.postMessage = postMessage
        self.timeStamp = timeStamp
        self.posterId = posterId
        self.posterName = posterName
        self.hasImage = hasImage
    }
    
    
    
    // Getters / Setters
    func getPostMessage() -> String {
        return postMessage
    }
    
    func setPostMessage(PostMessage postMessage: String) {
        self.postMessage = postMessage
    }
    
    func getTimeStamp() -> Date {
        return timeStamp
    }
    
    func setTimeStamp(TimeStamp timeStamp: Date) {
        self.timeStamp = timeStamp
    }
    
    func getPosterId() -> String {
        return posterId
    }
    
    func setPosterId(PosterId posterId: String) {
        self.posterId = posterId
    }
    
    func getPosterName() -> String {
        return posterName
    }
    
    func setPosterName(PosterName posterName: String) {
        self.posterName = posterName
    }
    
    func getHasImage() -> Bool {
        return hasImage
    }
    
    func setHasImage(HasImage hasImage: Bool) {
        self.hasImage = hasImage
    }
    
    
    
    // Custom methods
    func getPostId() -> String {
        return posterId + timeStamp.toMillis()!.description
    }
}



extension Date {
    
    func toMillis() -> Int64! {
        return Int64(self.timeIntervalSince1970 * 1000)
    }
    
    init(millis: Int64) {
        self = Date(timeIntervalSince1970: TimeInterval(millis / 1000))
        self.addTimeInterval(TimeInterval(Double(millis % 1000) / 1000 ))
    }
}
